packages/oidc/src/common/negotiation.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+// FIXME: this is copy-pasted from @inrupt/solid-client-authn-core (as is all the rest of this directory).
+// Initially, it was aimed that @inrupt/oidc-client-ext would not depend on the core package, and only
+// on the underlying OIDC library, but it is beginning to be impractical. This should be resolved in a
+// future PR.
```

Comment:
Maybe also mention here what the resolution should be in that future PR, i.e. should it be consolidated into -core, or extracted into the individual packages?

Reply:
Well I wasn't sure what the resolution should be really :). My initial thought was to have `oidc-client-ext` simply depend on `solid-client-authn-core`, but I think there was a reason not to do that earlier, so I preferred not to say anything in the comment, and consolidate this properly later.

Reply:
Ah haha, yeah that makes it hard to note down the resolution :P